src/higher-order/with-design-layout-selector/index.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+import {
+	i18n, srcUrl, isPro,
+} from 'stackable'
+import { applyBlockDesign } from '~stackable/util'
+import { getDesigns, getDesign } from '~stackable/design-library'
+import classnames from 'classnames'
+
+/**
+ * Wordpress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose'
+import {
+	Fragment, useEffect, useState,
+} from '@wordpress/element'
+import {
+	applyFilters, addAction, removeAction, doAction, hasFilter, addFilter, hasAction,
+} from '@wordpress/hooks'
+import { __ } from '@wordpress/i18n'
+import {
+	Placeholder, Icon, Button, ButtonGroup, BaseControl,
+} from '@wordpress/components'
+import {
+	dispatch, useSelect, select,
+} from '@wordpress/data'
+
+if ( ! hasFilter( 'stackable.with-design-layout-selector.switch-design-panel', 'switch-design-layout' ) ) {
+	addFilter( 'stackable.with-design-layout-selector.switch-design-panel', 'switch-design-layout', () => {
+		const selectedBlockId = select( 'core/block-editor' ).getSelectedBlockClientId()
+		const block = select( 'core/block-editor' ).getBlocksByClientId( selectedBlockId )
+
+		return (
+			<BaseControl>
+				<h5 className="ugb-design-layout-selector__filter-title">{ __( 'Layouts and Preset Designs', i18n ) }</h5>
+				<div className="ugb-design-layout-selector__filter-content">
+					<Icon icon="info-outline" />
+					<p>{ __( 'You may switch your layout or choose a preset design from our Design Library here.', i18n ) }</p>
+				</div>
+				{ block && hasAction( `stackable.design-layout-selector.${ selectedBlockId }` ) && (
+					<div className="ugb-design-layout-selector__filter-button-group">
+						<Button
+							className="ugb-design-layout-selector__filter-button"
+							onClick={ () => {
+								doAction( `stackable.design-layout-selector.${ selectedBlockId }`, true )
+							} }
+							isSecondary
+							isLarge
+						>
+							{ __( 'Switch design/layout', i18n ) }
+						</Button>
+					</div>
+				) }
+			</BaseControl>
+		)
+	} )
+}
+
+const LayoutDesignSelectorItem = ( {
+	image,
+	className,
+	active,
+	label,
+	...otherProps
+} ) => {
+	const src = ! image ? '' :
+	            image.match( /https?:/i ) ? image :
+	            srcUrl ? `${ srcUrl }/${ image }` :
+	            image
+
+	const itemClassNames = classnames( 'ugb-design-layout-selector__item', {
+		[ className ]: className,
+	} )
+	const imgClassNames = classnames( 'ugb-design-layout-selector__image', {
+		active,
+		'is-premium': ! isPro && otherProps.plan === 'premium',
+	} )
+
+	return (
+		<div className={ itemClassNames } { ...otherProps }>
+			{ otherProps.plan && ! isPro && otherProps.plan !== 'free' && (
+				<span className="ugb-design-layout-selector__premium">{ otherProps.plan }</span>
+			) }
+			{ src && <img className={ imgClassNames } src={ src } alt={ label } /> }
```

Comment:
* Please wrap this in a `<button>` but don't include the label.
* Add a hover effect where the button will become a bit gray `#f0f0f0`, you can use `mix-blend-mode: darken` on the image to get rid of the white background. (please do this for the design buttons too)